dawn/src/driver-includes/cuda_verify.hpp

```diff
@@ -32,31 +33,166 @@ struct compute_error<AbsErrTag> {
   }
 };
 
-template <>
-struct compute_error<RelErrTag> {
-  static double __device__ impl(const double expected, const double actual) {
+template <typename FieldType>
+struct compute_error<RelErrTag, FieldType> {
+  static double __device__ impl(const FieldType expected, const FieldType actual) {
     double error = 0.;
     if(expected != actual) {
       error = fabs((expected - actual) / expected);
     }
```

Comment:
I think `(expected - actual)/expected` will not yield the desired result if `expected` and `actual` are both integers. In that case integer division applies, and they are only later casted to float because the result is used as an input to `fabs`.

One workaround would be
```
error = fabs(((double)expected - actual) / expected);
```

Maybe there is something more elegant ...